### PR TITLE
Remove !CPU_HAS_FP_SUPPORT code

### LINKED ISSUE #47708: How relevant is CPU_HAS_FP_SUPPORT?
I see there is a little bit of code that switches on this define here and there:
```
importer.cpp:12698:#if !CPU_HAS_FP_SUPPORT
lclvars.cpp:3584:#if CPU_HAS_FP_SUPPORT
lsra.cpp:1526:#if CPU_HAS_FP_SUPPORT
morph.cpp:12498:#if !CPU_HAS_FP_SUPPORT
morph.cpp:15497:#if !CPU_HAS_FP_SUPPORT
scopeinfo.cpp:380:#if !CPU_HAS_FP_SUPPORT
scopeinfo.cpp:414:#if CPU_HAS_FP_SUPPORT
```
Is that actually all that's needed to support porting to CPUs without FP support?
What is the purpose of this define?

FWIW, it is set to `1` for all currently supported targets.

### PR DESCRIPTION
**Pull request by BruceForstall**
CPU_HAS_FP_SUPPORT is always defined and we don't anticipate needing
the emulation support.

Fixes #47708

### CODE REVIEW

**Comment by BruceForstall** [comment]
@briansull PTAL
cc @dotnet/jit-contrib

**Review by echesakov — APPROVED** [review]
LGTM

File: src/coreclr/jit/morph.cpp [hunk 1]
```diff
@@ -15673,200 +15669,6 @@ GenTree* Compiler::fgRecognizeAndMorphBitwiseRotation(GenTree* tree)
...
-        /* Is this a binary operator? */
-
-        if (op2)
-        {
-            /* What kind of an operator do we have? */
-
-            switch (oper)
-            {
-                case GT_ADD:
-                    helper = CPX_R4_ADD;
```
Comment (echesakovMSFT) [inline src/coreclr/jit/morph.cpp:15731]: Are these (`CPX_R4_ADD`, `CPX_R4_SUB`) undefined symbols?

**Review by briansull — APPROVED** [review]
Looks Good,
 Also remove fgMorphToEmulatedFP function declaration in compiler.h

File: src/coreclr/jit/morph.cpp (hunk 1 above)
Comment (BruceForstall) [reply src/coreclr/jit/morph.cpp:15731]: I couldn't find them, so I don't maybe their definitions disappeared long ago.

**Pull request merged**